drag_and_drop_v2/drag_and_drop_v2.py

```diff
@@ -294,101 +294,71 @@ def studio_submit(self, submissions, suffix=''):
         }
 
     @XBlock.json_handler
-    def do_attempt(self, attempt, suffix=''):
-        item = self._get_item_definition(attempt['val'])
+    def drop_item(self, item_attempt, suffix=''):
+        self._validate_drop_item(item_attempt)
 
-        state = None
-        zone = None
-        feedback = item['feedback']['incorrect']
-        overall_feedback = None
-        is_correct = False
-
-        if self._is_attempt_correct(attempt):  # Student placed item in a correct zone
-            is_correct = True
-            feedback = item['feedback']['correct']
-            state = {
-                'zone': attempt['zone'],
-                'correct': True,
-                'x_percent': attempt['x_percent'],
-                'y_percent': attempt['y_percent'],
-            }
-
-        if state:
-            self.item_state[str(item['id'])] = state
-            zone = self._get_zone_by_uid(state['zone'])
+        if self.mode == self.ASSESSMENT_MODE:
+            return self._drop_item_assessment(item_attempt)
+        elif self.mode == self.STANDARD_MODE:
+            return self._drop_item_standard(item_attempt)
         else:
-            zone = self._get_zone_by_uid(attempt['zone'])
-        if not zone:
-            raise JsonHandlerError(400, "Item zone data invalid.")
+            raise JsonHandlerError(500, _("Unknown DnDv2 mode {mode} - course is misconfigured").format(self.mode))
 
-        if self._is_finished():
-            overall_feedback = self.data['feedback']['finish']
+    @XBlock.json_handler
+    def do_attempt(self, data, suffix=''):
+        self._validate_attempt()
 
-        # don't publish the grade if the student has already completed the problem
-        if not self.completed:
-            if self._is_finished():
-                self.completed = True
-            try:
-                self.runtime.publish(self, 'grade', {
-                    'value': self._get_grade(),
-                    'max_value': self.weight,
-                })
-            except NotImplementedError:
-                # Note, this publish method is unimplemented in Studio runtimes,
-                # so we have to figure that we're running in Studio for now
-                pass
+        self.num_attempts += 1
+        self._mark_complete_and_publish_grade()
 
-        self.runtime.publish(self, 'edx.drag_and_drop_v2.item.dropped', {
-            'item_id': item['id'],
-            'location': zone.get("title"),
-            'location_id': zone.get("uid"),
-            'is_correct': is_correct,
-        })
+        __, placed, correct = self._get_item_raw_stats()
+        placed_ids, correct_ids = set(placed), set(correct)
+        misplaced_ids = placed_ids - correct_ids
 
-        if self.mode == self.ASSESSMENT_MODE:
-            # In assessment mode we don't send any feedback on drop.
-            result = {}
+        feedback_msgs = [_('Correctly placed {correct_count} items, misplaced {misplaced_count} items.').format(
```

Comment:
Can we support pluralization? This returns "Correctly placed 1 items, misplaced 1 items.". `ngettext` might help.